### PR TITLE
Allow variadic arguments to `m.fragment`

### LINKED ISSUE #2275: Make `m.fragment` shadow `m`'s signature overloading logic
## Expected Behavior
`m.fragment`'s signature should be as tolerant as `m`, allowing for optional attributes, optional children, and the possibility of variadic children.

## Current Behavior
`m.fragment` throws unless it is given an attribute object as its first argument and an array-like as its second.

## Steps to Reproduce (for bugs)
| Invocation | Error |
| --- | --- |
| `m.fragment()` | TypeError: Cannot read property 'key' of undefined |
| `m.fragment({})` | TypeError: Cannot read property 'length' of undefined |
| `m.fragment([])` | TypeError: Cannot read property 'length' of undefined |
| `m.fragment(_, vnode, ....?)` | Silently fails to render anything |

### PR DESCRIPTION
**Pull request by isiahmeadows**
## Description

Basically, same idea as #2327, done almost at the same time. 😄 

## Motivation and Context

See #2327 - it includes a better writeup than this.

**Edit:** Fixes #2275.

**Edit:** Closes #2327.

## How Has This Been Tested?

Added new tests, ran existing ones.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation change

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] I have updated `docs/change-log.md`

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -1,5 +1,7 @@
 * text=auto
 /mithril.js binary
 /mithril.min.js binary
+/mithril.mjs binary
```
Comment (isiahmeadows) [inline .gitattributes]: This slip-in change was because #2325 missed it, and I forgot to add it in as part of the review.

**Comment by isiahmeadows** [comment]
@barneycarroll You beat me by like 15 seconds... 😄

**Comment by isiahmeadows** [comment]
@barneycarroll I incorporated your docs changes.

File: render/hyperscriptVnode.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+"use strict"
+
+var Vnode = require("../render/vnode")
+
+// Call via `hyperscriptVnode.apply(startOffset, arguments)`
```
[the remark this answers is not in the document]
Comment (isiahmeadows) [reply render/hyperscriptVnode.js]: So I don't have to create an array of arguments in the callees on a perf-critical path. It's a `this`-based performance hack.

In native ES6, I'd instead have it as `hyperscriptVnode(...args)`, where `...args` is the final parameter of each. But ES5 doesn't give me that luxury, and engines aren't nearly intelligent enough to 1. elide a useless `[].slice.call(arguments, 1)` call and 2. elide an `arguments` allocation when it's passed to another function only to be indexed.

[the remark this answers is not in the document]
Comment (barneycarroll) [reply render/hyperscriptVnode.js]: Gotcha. Any other scenario would be worse.

**Pull request merged**